src/inbox/mod_inbox_muc.erl

```diff
@@ -60,18 +58,19 @@ update_inbox_for_muc(
 is_allowed_affiliation(outcast) -> false;
 is_allowed_affiliation(_)       -> true.
 
--spec update_inbox_for_user(Direction, Host, Room, To, Packet) -> term() when
+-spec update_inbox_for_user(HostType, Direction, Room, To, Packet) -> term() when
+      HostType :: mongooseim:host_type(),
       Direction :: incoming | outgoing,
-      Host :: receiver_host(),
       Room :: room_bare_jid(),
       To :: receiver_bare_user_jid(),
       Packet :: packet().
-update_inbox_for_user(Direction, Host, Room, To, Packet) ->
+update_inbox_for_user(HostType, Direction, Room, To, Packet) ->
+    Host = To#jid.lserver,
     case {is_local_xmpp_host(Host), Direction} of
```

Comment:
Here we are checking if `Host` matches **any** host type, but below we are passing **the** host type that we got as the first argument. Do we still need the check? Is it possible to run this hook for a host type that does not match `Host`? Please double check that. If so, we should check if both host types are identical.